src/python/library/tritonclient/grpc/_infer_input.py

```diff
@@ -86,6 +87,36 @@ def shape(self):
         """
         return self._input.shape
 
+    def validate_data(self):
+        """Validate input has data and input shape matches input data.
+
+        Returns
+        -------
+        None
+        """
+        # Input must set only one of the following fields: '_raw_content',
+        # 'shared_memory_region' in '_input.parameters'
+        cnt = 0
+        cnt += self._raw_content != None
+        cnt += "shared_memory_region" in self._input.parameters
+        if cnt != 1:
+            return
+
+        # Skip due to trt reformat free tensor
+        if "shared_memory_region" in self._input.parameters:
+            return
+
+        # Not using shared memory
+        expected_num_elements = num_elements(self._input.shape)
+        data_num_elements = num_elements(self._data_shape)
+        if expected_num_elements != data_num_elements:
+            raise_error(
+                "input '{}' got unexpected elements count {}, expected {}".format(
```

Comment:
Can you also include the respective shapes in the error message as well?

something like:

`input 'XYZ' got unexpected elements count 8 (shape: 8,1), expected 16 (shape: 16,1)`

I think you are trying to keep supporting the case where a user might just want to call `set_shape()` with a different shape but same underlying data.

Reply:
Are shapes (8,2), (4,4) also valid?